feat(color): UI for 3.0

philmoz opened the pull request:
Adding this PR for the 3.0 UI branch so people can (hopefully) try it and give feedback.

First piece is a revised quick menu with two levels of navigation.
Top level is the same as the current quick menu.
When a menu item is selected that has a sub-menu is selected, the sub menu is shown.
Pressing ENTER switches navigation to the sub menu, RTN to go back to top level.

![screenshot_tx16s_24-11-26_16-56-40](https://github.com/user-attachments/assets/1f662c74-6b99-4760-81fa-381f29b6508a)

![screenshot_tx16s_24-11-26_16-56-45](https://github.com/user-attachments/assets/2c149c90-ee9c-4b42-9ce9-44065420b353)

richardclli commented:
How about vertical screens?  How this shows?

philmoz commented:
> How about vertical screens? How this shows in EL18?

Same as current - less columns, more rows.

![screenshot_el18_24-12-01_15-16-10](https://github.com/user-attachments/assets/4ccc46c2-08bb-4630-aa07-b35c2badbfd3)

philmoz commented:
Part 2 - remove tab page icons from Model and Radio setup menus.
- Move title to top bar to give more room for page content
- Add button to open quick menu (top left)
- Add button to close and return to main view (top right)

Only the radio and model setup menus are changed for now.
Pressing the quick menu button (top left) opens the quick menu with the current page selected.
The UI for the quick menu and close buttons is placeholder at the moment (pending better design).

![screenshot_tx16s_24-12-03_14-50-48](https://github.com/user-attachments/assets/29ea78eb-00c2-4a16-85f4-95b1b65847dd)
![screenshot_tx16s_24-12-03_14-50-54](https://github.com/user-attachments/assets/949e337a-b6ee-4fc7-89a7-9e9172cb9df3)

JimB40 commented:
Got your msg. :)
Right on the spot I can see one UI/UX fix  - there should be only one focus marker. 
That means that on top row you dim nonselected icons lea selected icon is highlighted but without a frame that serves as focus marker.
Will flash is today and give a test run.

philmoz commented:
Changes:
- SYS key opens quick menu
- Changed setup page header button UI
- Long press RTN closes quick menu (shortcut so you don't have to press RTN twice when a sub menu is open)

![screenshot_tx16s_24-12-09_08-07-24](https://github.com/user-attachments/assets/a407e2ed-db9d-480e-89a0-5e6308a3198e)

nhutlv01 commented:
- I prefer fullscreen popup for more space for the buttons
- Better blur background effect, please take a look at the iPhone group apps. We shouldn't see through that background to make the popup more focus
- Prefer different styles for main buttons and child buttons

dsolaray commented:
question. why don't you make colored icon sets instead of having them all-white ?
why are all app developers copying this design shit created by indians for M$ ?
do you imagine if all of your iphone apps icons were all white like this ?

JimB40 commented:
MANAGE MODELS
1. icon change

<img width="2790" height="580" alt="MANAGE MODELS" src="https://github.com/user-attachments/assets/f0799f1e-59b8-4851-813d-009e38bc78ea" />

JimB40 commented:
RADIO SETUP

1. Quick Menu 1st level group name change "Radio Settings" -> "RADIO SETUP"
2. Quick Menu 2nd level item name change "RADIO SETUP" -> "GENERAL SETTINGS"
3. Quick Menu item "TOOLS" removed from group
4. Quick Menu item "SD Browser" removed from group
5. Quick Menu item "THEMES" removed from group
6. Screen "HARDWARE->ANALOGS" need to be reworked. Topbar shows group names and Grup item name as any other screen. Instead of icons buttons "<" and ">" should be placed left to close button to allow navigation through 4 different analog monitor srcreens

<img width="2790" height="3462" alt="RADIO SETUP" src="https://github.com/user-attachments/assets/80ac7e52-259d-400f-b93e-85d7b87f34e9" />

JimB40 commented:
UI SETUP

1.  Quick Menu 1st level group name change "Screens Settings" -> "UI SETUP"
2. Quick Menu item "THEMES" added to group (moved from RADIO SETUP)
3. Quick Menu 2nd level item name change "User Interface" -> "TOP BAR"
4. "TOP BAR" (prevoiusly User Interface" screen needs to be reworked. Delete theme section as theme can be set via "THEMES" item
5. Quick Menu 2nd level items name change "Main View X" -> "SCREEN X"
6. Quick Menu 2nd level items name change "Add main view" -> "ADD SCREEN"

<img width="2790" height="2277" alt="UI SETUP" src="https://github.com/user-attachments/assets/7fa4196f-19f2-4650-a666-b68de2e0bc15" />

JimB40 commented:
TOOLS

1. New group called "TOOLS".
2. Quick Menu item "LUA APPS" added to group. Moved from RADIO SETUP called "Tools". New Icon.
3. Quick Menu item "STORAGE BROWSED" added to group. Moved from RADIO SETUP called "SD CARD"
4. Quick Menu item "FLIGHT RESET" added to group. Moved from Quick Menu 1st Level called "Reset flight". New Icon
5. Quick Menu item "CHANNEL MONITOR" added to group. Moved from Quick Menu 1st Level group called "Channel Monitor".
6. "CHANNEL MONITOR" screen needs to be reworked. Topbar shows group name "MONITOR" and Grup item name "CHANNELS 1-8" as any other screen. Buttons "<" and ">" should be placed left to close button to allow navigation through 4 different monitor srcreens "CHANNELS 1-8", "CHANNEL 9-16", "CHANNELS "17-25", "CHANNELS 25-32"
7. Quick Menu item "LOGIC SW MONITOR" added to group. Moved from Quick Menu "Channel Monitor->LOGICAL SWITSHES MONITOR ".
8. Quick Menu item "SPECTRUM ANALYSER" added to group. Moved from "Radio settings->Tools->Spektrum Analyser".
9. Quick Menu item "STATISTIC" added to group. Moved from "Statistics->Statictics"
10. Quick Menu item "DEBUG" added to group. Moved from "Statistics->Debug"

 
<img width="2790" height="3540" alt="TOOLS" src="https://github.com/user-attachments/assets/9dcdd3a0-2b30-42b0-b70f-0abde37572b3" />

philmoz commented:
@JimB40 Comments on the above changes:
- please provide .png files for the new icons (manage models, apps, etc). Three sizes are needed - 24x24px, 30x30px and 45x45px.
- please provide visual design for the Hardware->Analogs and Channel Monitor pages.
- 'Apps' is not just for Lua, there are built in tools that belong here (ghost config, spectrum analyser, etc).
- Spectrum analyser does not belong as a separate item (see previous comment).
- 'Model notes' is missing.

philmoz commented:
I have updated the PR with the revised menu structure.

The icon size has been increased to accommodate a larger font for the text under the icons.
This means that the top level is limited to 6 icons max and the sub level to 12 icons max (per top level item).

Outstanding issues:
- [ ] new 'Manage models' icon PNG files (3 sizes - 24x24px, 30x30px and 45x45px)
- [ ] new 'Apps' icon PNG files (3 sizes as above)
- [ ] new 'Flight reset' icon PNG files (3 sizes as above)
- [ ] 'Model notes' not in menu structure (currently removed)
- [ ] 'About EdgeTX' not in menu structure (currently removed)
- [ ] Visual design for 'Channel monitor' to remove tabs
- [ ] Visual design for 'Hardware / Analogs' to remove tabs
- [ ] Update icon/menu page text strings to match design images above.

<img width="480" height="272" alt="screenshot_tx16s_25-08-01_15-56-16" src="https://github.com/user-attachments/assets/8af8122d-90aa-44d6-9df6-1b2f0291e566" />

<img width="320" height="480" alt="screenshot_el18_25-08-01_15-56-05" src="https://github.com/user-attachments/assets/1055d4ce-59cc-48d3-9eb7-64b776c300a7" />

JimB40 commented:
Icons pack.

comparing to posted design
- "Manage Models" new
- "Radio Setup" new
- "Radio Setup > General Settings" changed
- "Model Setup > General Settings" changed (same as Radio)
- "Tools > Apps" new
- "Filght Reset" new

3 icons for TopBar navigation
- "Close" smaller with frame
- "Next" for LVGL button to navigate "Analogs" and/or "Channel Monitor"
- "Prev"  for LVGL button to navigate "Analogs" and/or "Channel Monitor"

[2025-08-01 3.0 Icons.zip](https://github.com/user-attachments/files/21548933/2025-08-01.3.0.Icons.zip)

JimB40 commented:
Hardware > Analogs

<img width="480" height="272" alt="screen-2000-09-18-063414" src="https://github.com/user-attachments/assets/01f90021-e44a-419e-8638-77b7c7844dac" />
<img width="480" height="272" alt="screen-2000-09-18-063424" src="https://github.com/user-attachments/assets/4aff3208-9264-493d-891f-434c3ba36a9f" />
<img width="480" height="272" alt="screen-2000-09-18-063430" src="https://github.com/user-attachments/assets/e115a82d-37cd-41ef-b65f-48108b29da03" />
<img width="480" height="272" alt="screen-2000-09-18-063436" src="https://github.com/user-attachments/assets/46af0df0-a62c-4298-ae8c-96b3b26cde09" />

philmoz commented:
Icons have been replaced. Channel Monitor and Analogs screen UI updated.

On landscape layouts I moved the title line for Channel Monitor and Analog into the top bar for consistency and to increase space for page content.

<img width="480" height="272" alt="screenshot_tx16s_25-08-02_14-15-20" src="https://github.com/user-attachments/assets/dad78f33-cd55-4106-a758-1318306e0fdc" />
<img width="480" height="272" alt="screenshot_tx16s_25-08-02_14-15-31" src="https://github.com/user-attachments/assets/c79ad2d5-7a0e-4f7e-8f44-de715d136008" />

On portrait layouts I had to keep the extra line for the title due to space constriants.

<img width="320" height="480" alt="screenshot_el18_25-08-02_14-19-43" src="https://github.com/user-attachments/assets/bda4c712-7bee-4e56-921b-b6c5ca121e91" />
<img width="320" height="480" alt="screenshot_el18_25-08-02_14-19-54" src="https://github.com/user-attachments/assets/d60cd1d0-274a-4526-8f77-0095769de40f" />

philmoz commented:
Merged the 'About' QR code into the 'Version' page. I changed the name to 'About EdgeTX'.

For portrait layout radios I have changed the quick menu top level to a vertical layout as I think it looks cleaner.
The sub menus still layout by row.
@JimB40 what do you think?

<img width="320" height="480" alt="screenshot_el18_25-08-02_18-56-15" src="https://github.com/user-attachments/assets/82e2e187-6dd5-4d60-a750-022c4497787e" />
<img width="320" height="480" alt="screenshot_el18_25-08-02_18-56-27" src="https://github.com/user-attachments/assets/b0fb3f6c-9283-49dd-8be0-3fe65f613880" />

philmoz commented:
@JimB40 - I have updated the PR with all the new icons from the zip file. Please check that all are correct. In future it would help if you use the existing file names so I don't have to manually update each one.

JimB40 commented:
> In future it would help if you use the existing file names so I don't have to manually update each one.
Already was trying to find a way in Figma to get objects custom name for export. So far no luck.

JimB40 commented:
Okay so Typography for navigation tree

My go to solution for menu is always capitals for better readability. As we have in all other menus (lists)
Navigation tree names appear in QM and TopBar, so I can see that using capitals will easily exeed alowed space in QM and TopBar for vertical LCD. No mentioning that in some cases some of TopBar right side area is occupied by aux buttons 

Key assumptions
1. Due to above data space constrains we keep 1st and 2nd level Navigation Tree names consisting max 2 words

2. Names are stored internaly in low case format with 1st letter captalized. ie "Manage Models",  "Storage", "Global Variables"

3. We may try to display 1st and 2nd level names just on horizontal LCD TopBar  converting stored name via uppecase(name) but let's start with low case 1st letter capitalized it may look okay.

3. In QM with 2 words names we try to put them in two lines to provide separation from frame & readbility
Examples of good separation

<img width="480" height="272" alt="screen-2000-09-20-004443" src="https://github.com/user-attachments/assets/f160eb58-bcb1-4dab-bfce-28c209f73ffc" />

<img width="480" height="272" alt="screen-2000-09-20-005234" src="https://github.com/user-attachments/assets/84824685-e4b4-48b0-95b4-5cd2ce7a0411" />

Examples of bad separation

<img width="480" height="272" alt="screen-2000-09-20-005513" src="https://github.com/user-attachments/assets/4e479b14-75a1-4ddb-ba32-d3359f768a5d" />

<img width="480" height="272" alt="screen-2000-09-20-005518" src="https://github.com/user-attachments/assets/49b6f80f-35e4-4f06-aa30-0dad15d6cf26" />

<img width="480" height="272" alt="screen-2000-09-20-004456" src="https://github.com/user-attachments/assets/496c70a5-45b8-4cdb-a96a-37555ce71600" />

This would need you to write some smart functiond that will split wide 2 words names into 2 lines in QM.
or maybe store names with special character between words that will be replaced with /br in QM and with space in case or TopBar. Just ideas if you find it to difficult we will try differen solution (like hilighting icon backgound that will take less space)

JimB40 commented:
QM Naming review for last arifact (#b3c330e)

1. Radio Setup > Radio Setup   change    Radio Setup > General Settings

2. Model Setup > Model Setup   change    Model Setup > General Settings
so it will appear in TopBar as
Radio Setup
General Settings
not 
Radio Setup
Radio Setup

3. UI Settings   change   UI Setup
1st Level uses "Setup" while 2nd level "Settings"

4. Model Setup > Heli Setup   change  Model Setup > Heli Settings
same as in point 3

5. UI Setup > User Interface    change    UI Setup > Top Bar
that's content on destination screen when duplicated theme selection is ditched.

6. Tools > Channel Monitor   change   Tools > Channels Monitor
plurar form

Rest seems proper.

philmoz commented:
Feedback on discord (General channel) - https://discord.com/channels/839849772864503828/839849772864503832/1401503862891876473

philmoz commented:
> 1. Radio Setup > Radio Setup   change    Radio Setup > General Settings
> 2. Model Setup > Model Setup   change    Model Setup > General Settings
> 3. UI Settings   change   UI Setup
> 4. Model Setup > Heli Setup   change  Model Setup > Heli Settings

Done.

> 5. UI Setup > User Interface    change    UI Setup > Top Bar
>    that's content on destination screen when duplicated theme selection is ditched.

Do you want to remove theme selection from this page?

> 6. Tools > Channel Monitor   change   Tools > Channels Monitor

That's not normal English grammar - a 'box of apples' is an 'apple box', not an 'apples box'. It is also described as 'Channel Monitor' in the documentation (https://manual.edgetx.org/color-radios/channel-monitor).

philmoz commented:
I have added code to split the menu titles to two lines by replacing spaces with newline characters ('\n'). It does not do this if the 2nd word is 1 or 2 characters long (Screen 1 - Screen 10 is not split).

This assumes that menu titles are one or two words. I have not checked all the language files so there may be some that break. It may also break on RTL languages.

JimB40 commented:
> > 5. UI Setup > User Interface    change    UI Setup > Top Bar
> >    that's content on destination screen when duplicated theme selection is ditched.
> 
> Do you want to remove theme selection from this page?

Yes. Theme can be set as active in "UI Setup > Themes". Redundant.

> 
> > 6. Tools > Channel Monitor   change   Tools > Channels Monitor
> 
> That's not normal English grammar - a 'box of apples' is an 'apple box', not an 'apples box'. It is also described as 'Channel Monitor' in the documentation (https://manual.edgetx.org/color-radios/channel-monitor).

Okay, you're native.

gonzalo-bulnes commented:
> Key assumptions
> 
> 1. Due to above data space constrains we keep 1st and 2nd level Navigation Tree names consisting max 2 words
> 2. Names are stored internaly in low case format with 1st letter captalized. ie "Manage Models",  "Storage", "Global Variables"
> 3. We may try to display 1st and 2nd level names just on horizontal LCD TopBar  converting stored name via uppecase(name) but let's start with low case 1st letter capitalized it may look okay.
> 

Hi folks, you may already be considering the following limitations of the above assumptions, if so, please don't feel pressure to justify decisions that you've already discussed at length! I mention them just in case it's useful as an enthusiatic EdgeTX user, but I'm very aware I'm most likely missing a lot of context.

Since the EdgeTX UI is used with a number of languages, I'd keep in mind that:

- while the number of words can be relatively controlled in English, it may be harder in other languages (e.g. German often relies in compound words rather than groups of words, those may be difficult to split in two lines; Spanish is way less compact than English for technical terms)
- relying on automated capitalization (style layer, database layer) can be trickier than letting copywriters decide on the best capitalization. (German comes to mind again, because capitals do have gramatical meaning.)

JimB40 commented:
Did fe hour testing. 
Adding labels obscured almost whole screen so tapping outside QM to close it on key-less radio became more problematic.
What you think @philmoz  about obscuring whole screen?
- will give som more whitespaces
- allow to separete visually 1st level and 2nd level
- logo will serve as close button for key-less radios 
 
<img width="480" height="272" alt="UI 3 0 QM Horizontal" src="https://github.com/user-attachments/assets/bb51e21f-9db4-4a4f-a97e-7aba170b8627" />
<img width="320" height="480" alt="UI 3 0 QM Vertical" src="https://github.com/user-attachments/assets/ea92a998-2da0-4bfb-9b25-e28ea9ce98da" />

philmoz commented:
@JimB40 Full screen may be better, I can space things to use the available height on landscape radios.

I would suggest adding the close button as well. If you want to add the Edge TX title please provide PNG file for it (at .8x, 1x and 1.5x scaling).

<img width="480" height="272" alt="screenshot_tx16s_25-08-04_09-29-21" src="https://github.com/user-attachments/assets/5c2d621f-2db0-4264-a01e-7e0ff2eb87df" />

breavyn commented:
I prefer more opacity and full screen as these radios are often used under challenging lighting conditions.

dsolaray commented:
> What you think @philmoz about obscuring whole screen?
> More opacity

I personally prefer white backgrounds but it looks better now blacked out (full opacity)
My question is why don't EdgeTx devs like the colorized icons ?

<img width="480" height="272" alt="color1" src="https://github.com/user-attachments/assets/6b4e17eb-e979-420e-a5ef-d29a9b3eed49" />
<img width="480" height="272" alt="color2" src="https://github.com/user-attachments/assets/75d6ae91-1c6f-4743-90ed-fd67fb40c2c1" />

gagarinlg commented:
I prefer the higher contrast from the B&W icons

dsolaray commented:
> I prefer the higher contrast from the B&W icons

Multiple choice of skins so we can please everyone. 👪 👪 👪
the reason I hate the modern Windowzz 💻 or smartphones📱apps it's because these uncolored icons are a waste of time and hard to differentiate at first glance.
When applying colors the brain will memorize the function without having to read the bottom text

philmoz commented:
> > I prefer the higher contrast from the B&W icons
> 
> Multiple choice of skins so we can please everyone. 👪 👪 👪 the reason I hate the modern Windowzz 💻 or smartphones📱apps it's because these uncolored icons are a waste of time and hard to differentiate at first glance. When applying colors the brain will memorize the function without having to read the bottom text

We don't support coloured icons in the firmware and we don't have the manpower to support skins.

Gerold68 commented:
I have tested one built yesterday. Now I do not see the artifacts for last fix.

gismo2004 commented:
I am using it since quite some time on V16 --> very happy with it with current status. 

My personal remarks:

- The top 5 row looks a bit misaligned to me as it is left aligned, and the right empty space looks somewhat strange. If it was centered, it would feel a bit more decoupled from the sub menu symbols. Maybe also/or a separator line/bar between top menu and submenu would be helpful for that. 
- Currently, it's hard to see which top menu is currently selected when in sunlight. (that might be just me)  A dark gray square around the selected item would be helpful and can be differentiated to the currently selected submenu item. (again a separator bar would help)
- If the menu is opened, there is no "one touch close without changes" function. Maybe touching empty space to close the menu?

gismo2004 commented:
> > * If the menu is opened, there is no "one touch close without changes" function. Maybe touching empty space to close the menu?
> 
> The quick menu can be quickly closed by:
> 
> * presssing SYS key
> * long pressing RTN key
> * touching the top bar (where the EdgeTX logo sits)

I was aware of the buttons, but I was talking about “touch”. However, this is interesting as this not even happened a single time by accident! Now that I know it, it's kind of NOT the “quick one touch close” thing on my V16 :-) (see video: [PXL_20250808_045322536.TS.mp4.zip](https://github.com/user-attachments/files/21677106/PXL_20250808_045322536.TS.mp4.zip)). 

Might be better on other radios though...

JimB40 commented:
> * The top 5 row looks a bit misaligned to me as it is left aligned, and the right empty space looks somewhat strange. If it was centered, it would feel a bit more decoupled from the sub menu symbols. Maybe also/or a separator line/bar between top menu and submenu would be helpful for that.

Last 6th  top row space is for "Favourites" group which is not implemented yet. 
Sepatation top row and  bottom rows was initial idea. With labels underneath icons there is no place for separation.
We need to find other solution  

> * Currently, it's hard to see which top menu is currently selected when in sunlight. (that might be just me)  A dark gray square around the selected item would be helpful and can be differentiated to the currently selected submenu item. (again a separator bar would help)

Inversion might be way to go to max contrast radio

> * If the menu is opened, there is no "one touch close without changes" function. Maybe touching empty space to close the menu?

Yes thats right. With labels underneath icons logo area is 24pc instead of notmal 34 px that makes touch challenging especially with thumb. 

As above @philmoz thinking about switching to inversion as seselection that will save 4 pixels fram on 480x272px

but for now we can bit improve 
With 70px tile height math is:
- top logo area 34 px
- space 6 px
- 1 st row 70px
- space separation 14px
- 2nd row 70px
- space 2 px
- 3rd row 70px
- bottom space 6px 
34+6+70+14+70+2+70+6 = 272

philmoz commented:
Latest update (once it finishes building) inverts the buttons for focus.

philmoz commented:
> but for now we can bit improve With 70px tile height math is:
> 
> * top logo area 34 px
> * space 6 px
> * 1 st row 70px
> * space separation 14px
> * 2nd row 70px
> * space 2 px
> * 3rd row 70px
> * bottom space 6px
>   34+6+70+14+70+2+70+6 = 272

I have increased the top bar button size to 32px (standard size).
Currently the buttons are set out in a flex container with the same spacing between icons both vertically and horizontally.

<img width="600" height="400" alt="screenshot_tx16s_25-08-08_18-56-44" src="https://github.com/user-attachments/assets/2903f6e1-9755-4d7d-b161-f3b216eb7e3a" />

gismo2004 commented:
> Latest update (once it finishes building) inverts the buttons for focus.

Sometimes, when changing menus, the "old" menu button gets completely black and the "new" menu button is completely white. This does not happen on every change, but sometimes.

gismo2004 commented:
> > > Latest update (once it finishes building) inverts the buttons for focus.
> > 
> > 
> > Sometimes, when changing menus, the "old" menu button gets completely black and the "new" menu button is completely white before the menu appears. This does not happen on every change, but sometimes.
> 

> Can you isolate steps to make it happen?

Not really, but seems to be a CPUload thing, as it happens more often when changing to mixer tab. 

[PXL_20250808_093906187.TS.mp4.zip](https://github.com/user-attachments/files/21680991/PXL_20250808_093906187.TS.mp4.zip)

JimB40 commented:
Layout with above given math 

<img width="1003" height="606" alt="Screenshot 2025-08-08 at 11 44 15" src="https://github.com/user-attachments/assets/e246b11d-e9e7-42df-b4c5-577c7147d076" />

If contrast ratio is priority, instead of dimming icons we have leave frame as selection mark.
Then 
- frame shows fhat is/was selected
- inversion shows what is focused to select

gismo2004 commented:
> Adjust focus state logic.

I have tested it, but does not fix the issue --> proceed on discord to not spam this PR?

philmoz commented:
> If contrast ratio is priority, instead of dimming icons we have leave frame as selection mark. Then
> 
> * frame shows fhat is/was selected
> * inversion shows what is focused to select

No - that's too complicated to manage.

breavyn commented:
What do you think about using the page buttons to cycle the top level of the quick menu? This would be an alternative to pressing return, scrolling, enter, when in a sub menu.

pfeerick commented:
LGTM... prior issues raised are gone. 

Only issues going forward that come to mind are 
 - use of the SYS key for the quick menu, given that is now two hard key buttons assigned to that - ENTER and SYS, with the loss of hard key shortcut to Tools (now "Apps)?
- Manage Models behaves differently to every other icon on the top row
- Make use of the page keys (when present) to move between top icons/groups
- Later down the track, this will probably want to be themeable (colour tint)

philmoz commented:
> * use of the SYS key for the quick menu, given that is now two hard key buttons assigned to that - ENTER and SYS, with the loss of hard key shortcut to Tools (now "Apps)?

Quick menu requires a dedicated key.
ENTER only works in the main view, when you are in any of the menu pages SYS is used to open the quick menu.
